Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

_[repo-dirty-checker/repo-dirty-checker, line 27 [r1]](https://reviewable.io:443/reviews/trojsten/sysadmin-script-collection/6#-KClsLZ99EE9NxVP7mkS:-KCm49EirsE0J5q8aRms:1029515265) ([raw file](https://github.com/trojsten/sysadmin-script-collection/blob/8b448236f567e481cee42a7576a27970254c4c09/repo-dirty-checker/repo-dirty-checker#L27)):_
Uz viem preco som ich odstranil. ShellCheck mi vypisoval: "In POSIX sh, 'local' is undefined. [SC2039]". Pozri si [SC2039](https://github.com/koalaman/shellcheck/wiki/SC2039) pre blizsie info.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/trojsten/sysadmin-script-collection/6)_

<!-- Sent from Reviewable.io -->

--- a/repo-dirty-checker/repo-dirty-checker
+++ b/repo-dirty-checker/repo-dirty-checker
@@ -7,6 +7,7 @@ CONFIG="/usr/local/etc/repo-dirty-checker.conf"
 
 DATA_DIR="/var/tmp/repo-dirty-checker"
 
+# ShellCheck directive, DO NOT REMOVE!
 # shellcheck source=repo-dirty-checker.conf
 . $CONFIG
 

--- a/repo-dirty-checker/repo-dirty-checker
+++ b/repo-dirty-checker/repo-dirty-checker
@@ -18,6 +18,17 @@ init_users_file() {
 	echo > "${1}"
 }
 
+print_warning() {
+	wall <<-END
+	Warning: uncommited changes detected in ${1}. Please clean up!
+	END
+}
+
+send_mail() {
+	# TODO(kuci): Implement mail sending
+	:
+}
+
 main() {
 	mkdir -p "${DATA_DIR}" || exit 1
 
@@ -44,9 +55,12 @@ main() {
 
 		if [ ! -z "$(git status --porcelain)" ]; then
 			run="$(cat "${run_file}")"
+			users="$(cat "${users_file}")"
 
 			if [ "${run}" -eq "0" ]; then
-				echo "$(who | cut -d ' ' -f 1 | sort -u)" > "${users_file}"
+				users="$(who | cut -d ' ' -f 1 | sort -u)"
+
+				echo "${users}" > "${users_file}"
 			fi
 
 			run="$(expr "${run}" + 1)"
@@ -57,9 +71,7 @@ main() {
 				runs_after="$(expr "${run}" - "${WALL_AFTER}")"
 
 				if [ "$(expr "${runs_after}" % "${WALL_PERIOD}")" -eq "0" ]; then
-					wall <<-END
-					Warning: uncommited changes detected in ${dir}. Please clean up!
-					END
+					print_warning "${dir}" "${users}"
 				fi
 			fi
 
@@ -67,8 +79,7 @@ main() {
 				runs_after="$(expr "${run}" - "${MAIL_AFTER}")"
 
 				if [ "$(expr "${runs_after}" % "${MAIL_PERIOD}")" -eq "0" ]; then
-					# Send mail
-					:
+					send_mail "${dir}" "${users}"
 				fi
 			fi
 		else

